From bc298dfc5642f28291bc6fef02ca038b621f70a8 Mon Sep 17 00:00:00 2001
From: Reilly Brogan <reilly@reillybrogan.com>
Date: Wed, 25 Oct 2023 12:02:27 -0500
Subject: [PATCH] solus: Don't automatically compress modules

If CONFIG_MODULE_COMPRESS_ZSTD is set then the kernel will automatically compress modules when `make modules_install` is called. This is not ideal as we would prefer to handle uncompressed modules and would need to add an additional decompression step only to re-compress them later.

However, if CONFIG_MODULE_COMPRESS_ZSTD is not set then the kernel is not compiled with built-in module decompression support. This is less efficient for our purposes too since we'd rather the kernel do module decompression rather than delegating it to user-space.

So instead let's just force the `modules_install` target to skip compression entirely. This keeps kernel decompression support and allows us to work with uncompressed kernel modules
---
 scripts/Makefile.modinst | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/scripts/Makefile.modinst b/scripts/Makefile.modinst
index 0afd75472679..10592f74e3ed 100644
--- a/scripts/Makefile.modinst
+++ b/scripts/Makefile.modinst
@@ -53,7 +53,7 @@ $(foreach x, % :, $(if $(findstring $x, $(dst)), \
 suffix-y				:=
 suffix-$(CONFIG_MODULE_COMPRESS_GZIP)	:= .gz
 suffix-$(CONFIG_MODULE_COMPRESS_XZ)	:= .xz
-suffix-$(CONFIG_MODULE_COMPRESS_ZSTD)	:= .zst
+suffix-$(CONFIG_MODULE_COMPRESS_ZSTD)	:=
 
 modules := $(patsubst $(extmod_prefix)%.o, $(dst)/%.ko$(suffix-y), $(modules))
 install-$(CONFIG_MODULES) += $(modules)
-- 
2.42.0

